flink-table/flink-table-common/src/main/java/org/apache/flink/table/sinks/PartitionableTableSink.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.sinks;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An abstract class with trait about partitionable table sink. This is mainly used for
+ * static partitions. For sql statement:
+ * <pre>
+ * <code>
+ * INSERT INTO A PARTITION(a='ab', b='cd') select c, d from B
+ * </code>
+ * </pre>
+ * We Assume the A has partition columns as &lt;a&gt;, &lt;b&gt;, &lt;c&gt;.
+ * The columns &lt;a&gt; and &lt;b&gt; are called static partition columns, while c is called
```

Comment:
I think we should give the definition of table `A` if we intend to offer a valid example. It's true the dynamic column should appear last, but the column names in `SELECT` don't have to be the same as the column names in the destination table -- so it's hard to tell w/o a DDL :)

Reply:
As partition fields are part of the schema, the table sink doesn't need to know whether it is a static partition or dynamic partition, and even doesn't need to know the specific static partition, it can be get from records. So I removed the comments about dynamic partition and `setStaticPartitions(...)` method.